### ISSUE TITLE
Announce <a> links with target="_blank" attribute as "Opens in a new tab"

### ISSUE THREAD

**Issue by zackpyle** [nvaccess/nvda#14875]
### Is your feature request related to a problem? Please describe.
If a developer doesn't add 'screen reader only' text announcing it opens in a new tab for a link with a target="_blank" attribute, then the user won't know that a new tab was opened.

### Describe the solution you'd like
NVDA should announce that the link opens in a new tab when a <a> tag has a target="_blank" attribute. For WCAG compliance, links with target="_blank" will be failed if they don’t have a visual and auditory warning on them telling the user they will open a new tab. If we can get screen reader softwares on board to announce these links automatically, that reduces the burden on developers to create accessible websites when this should be built into the screen reader software.

So if this were fixed by screen reader software, then adding a pseudo element to [target="_blank"]::after would be super easy.

### Describe alternatives you've considered
**Other screen readers have similar problems too.**
JAWS 2022.2207.25: No indication link opens in new window before clicking, or that one has opened after clicking link.
VoiceOver macOS 12.5: No indication link opens in new window before clicking, but Back button goes back to previous page.
And then NVDA 2022.2: No indication link opens in new window before clicking, or that one has opened after clicking link.

https://www.powermapper.com/tests/screen-readers/navigation/a-target-blank/

### Additional context
Currently the burden is on each individual web developer. But this solution would help hundreds of thousands of users with a simple change.

**Comment by seanbudd** [comment]
Closing this as not planned

Sighted users also experience this problem. It is up to the web author to visually indicate that a link opens in a new tab via an icon for sighted users and via a label or description for screen reader users.

**Comment by zackpyle** [comment]
But why not help hundreds of thousands of users instead of relying on individual developers to do this when NVDA could solve it in one fell swoop?

As stated in the feature request:

For WCAG compliance, links with target="_blank" will be failed if they don’t have a visual and auditory warning on them telling the user they will open a new tab. If we can get screen reader softwares on board to announce these links automatically, that reduces the burden on developers to create accessible websites when this should be built into the screen reader software.

**Sighted users can just have an icon in a pseudo element. But it's a violation to have that content in a pseudo element for screen readers. So if this were fixed by screen reader software, then adding a visual indicator with a pseudo element to [target="_blank"]::after would be SUPER easy.**

**Comment by futekov** [comment]
Respectfully, this needs to be reopened. I agree that "It is up to the web author to visually indicate that a link opens in a new tab via an icon for sighted users and via a label or description for screen reader users." and we already have an attribute for that - `target`. It even says so in the [WCAG guidelines](https://www.w3.org/WAI/WCAG21/Techniques/html/H83.html#:~:text=Use%20of%20the%20target%20attribute%20provides%20an%20unambiguously%20machine%2Dreadable%20indication%20that%20a%20new%20window%20will%20open.) - "Use of the target attribute provides an unambiguously machine-readable indication that a new window will open".

The `target` attribute can be used for screen readers AND for visual hints. Making NVDA respect this "unambiguous" machine-readable declaration of course does not cancel out the web author's responsibility to visually indicate a link (which can be done by adding as an SVG, with CSS pseudo element, etc).

**Comment by zackpyle** [comment]
Well said! 🙌🏻

**Comment by ramoncorominas** [comment]
Having a target="_blank" in an anchor does not guarantee that the link will really open in a new tab or window. Sometimes developers use Javascript to read those attributes and use them to create custom behaviours, such as opening the link contents within a dialog in the same page (for example, I've seen this for "legal terms" or "privacy policy" links).

I'm not saying that this is good practice, but it can be confusing and problematic if you automatically add an indication telling that a new tab/window will open if that tab/window is not finally opened. The user could close the "new" tab thinking he will return to the previous window, but since the window has not changed then the user would indeed close the original window.

**Comment by zackpyle** [comment]
@ramoncorominas Well that would be a direct violation of WCAG guidelines as @futekov mentioned. So I don't think that is a case that needs to be considered. [Use of the target attribute provides an unambiguously machine-readable indication that a new window will open.](https://www.w3.org/WAI/WCAG21/Techniques/html/H83.html#:~:text=Use%20of%20the%20target%20attribute%20provides%20an%20unambiguously%20machine%2Dreadable%20indication%20that%20a%20new%20window%20will%20open.)

**Comment by ramoncorominas** [comment]
> Well that would be a direct violation of WCAG guidelines as @futekov mentioned. So I don't think that is a case that needs to be considered.

There are thousands of direct violations of WCAG that cannot (and must not) be solved in the screen reader side. For example, using an image without an `alt` is a direct violation of WCAG and of course we cannot pretend that the screen reader inserts a description trying to solve that violation.


> [Use of the target attribute provides an unambiguously machine-readable indication that a new window will open.]

As I've previously explained, that is more a desire than a reality. Since the default behaviour can be overwritten that statement is simply not true. In any case, techniques are non-normative, and you are omitting the rest of the tecnique description, that is clearly aimed at providing the information about the new window in the text of the link itself, which would become redundant if the screen reader also adds this announcement.

In addition, the `target` attribute can be set to other values such as "_self" or to the name of an existing iframe; in both cases the browser will not open a new tab/window, but stay in the current one. Moreover, even if a new tab/window could be opened, you don't know in advance how the user has configured this behaviour to open these links in new tabs or windows, so you cannot give a reliable message: "new window"?, "new tab"?, "new tab or window"?. It is also possible that some browsers could be configured to completely avoid opening new windows or tabs unless they are forced by the user.

Lastly, even if the feature were desirable, I think it has some implementation issues. I guess that The `target` attribute is not exposed in any property of the accessibility APIs, so the screen reader would have to obtain it by other means, which I am not sure is even possible. If you want this feature I think you should better ask the browser vendors to implement it, since they have more control over the real behaviour and can expose that in some kind of API or simply add the proper text to the links.

**Comment by zackpyle** [comment]
I disagree with the majority of your opinions here 🤷🏼‍♂️

**Comment by RYJASM** [comment]
You (ramon) certainly have some HOT TAKES that ignore the entire purpose of the attribute by talking about what if, non-standard overrides and reliance on ambigous open ended solutions when there is already a native attribute that controls link functionality that would solve the accessibility issue.
